From 164bbcc3b414c639ea916a202f74238029c828d6 Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Wed, 7 Jun 2023 11:46:44 +0200
Subject: [PATCH 0367/1141] clk: stm32mp25: set gicv2m clock as critical

GIC driver support does not allow errors for specific gic
initialization so it can't support DEFER_PROBE.

Proposed solution is to mark the gicv2m clock critical in RCC
drivers.

Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Change-Id: I5a28fbbb4e876f99a36ef8925b524fd1a0bd1c29
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/311274
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 drivers/clk/stm32/clk-stm32mp25.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/clk/stm32/clk-stm32mp25.c b/drivers/clk/stm32/clk-stm32mp25.c
index 229b079f53d9..d423b38173ff 100644
--- a/drivers/clk/stm32/clk-stm32mp25.c
+++ b/drivers/clk/stm32/clk-stm32mp25.c
@@ -709,7 +709,7 @@ static CLK_STM32_GATE(ck_icn_p_fdcan, "ck_icn_apb2", 0, GATE_FDCAN);
 static CLK_STM32_GATE(ck_ker_fdcan, "ck_flexgen_26", 0, GATE_FDCAN);
 
 /* GICV2M */
-static CLK_STM32_GATE(ck_icn_p_gicv2m, "ck_icn_apb4", 0, GATE_GICV2M);
+static CLK_STM32_GATE(ck_icn_p_gicv2m, "ck_icn_apb4", CLK_IS_CRITICAL, GATE_GICV2M);
 
 /* GPU */
 static CLK_STM32_GATE(ck_icn_m_gpu, "ck_flexgen_59", 0, GATE_GPU);
-- 
2.39.2

